packages/chatbot-server-mongodb-public/src/tracing.ts

```diff
@@ -1,28 +1,46 @@
 import { AppConfig } from "mongodb-chatbot-server";
 import { SomeMessage, UserMessage } from "mongodb-rag-core";
 import { llmDoesNotKnowMessage } from "./systemPrompt";
+import { strict as assert } from "assert";
+import { ContextRelevancy, AnswerRelevancy, Faithfulness } from "autoevals";
+import { traced } from "mongodb-rag-core/braintrust";
 
 export const makeAddMessageToConversationUpdateTrace: (
-  k: number
+  k: number,
+  llmAsAJudge?: LlmAsAJudge
 ) => AppConfig["conversationsRouterConfig"]["addMessageToConversationUpdateTrace"] = (
-  k
-) =>
-  async function ({ traceId, addedMessages, logger }) {
+  k,
+  llmAsAJudge
+) => {
+  validatePercentToJudge(llmAsAJudge?.percentToJudge);
+
+  return async function ({ traceId, addedMessages, logger }) {
     const tracingData = extractTracingData(addedMessages);
+    const shouldJudge =
+      typeof llmAsAJudge?.percentToJudge === "number" &&
+      Math.random() < llmAsAJudge.percentToJudge;
+
     logger.updateSpan({
       id: traceId,
       tags: tracingData.tags,
       scores: {
-        RejectedQuery: tracingData.rejectQuery === true ? 1 : null,
-        VerifiedAnswer: tracingData.isVerifiedAnswer === true ? 1 : null,
-        LlmDoesNotKnow: tracingData.llmDoesNotKnow === true ? 1 : null,
-        [`RetrievedChunksOver${k}`]:
-          tracingData.isVerifiedAnswer !== true
-            ? tracingData.numRetrievedChunks / k
-            : null,
+        ...getTracingScores(tracingData, k),
+        ...(shouldJudge
+          ? await getLlmAsAJudgeScores(llmAsAJudge, tracingData)
+          : undefined),
       },
     });
   };
+};
+
+function validatePercentToJudge(percentToJudge?: number) {
```

Comment:
What if we combined this validator and the random number generation into a single function? Right now we're validating in two places (this function + `typeof llmAsAJudge?.percentToJudge === "number"` later)

e.g.

```ts
function shouldSample(percent: number | undefined): boolean {
  if (percentToJudge === undefined) {
    return false;
  }
  assert(
    percentToJudge >= 0 && percentToJudge <= 1,
    `percentToJudge must be between 0 and 1. Received: ${percentToJudge}`,
  );
  return Math.random() < percentToJudge;
}
```

Reply:
more elegant will do